book/website/reproducible-research/renv/renv-virtualmachine.md

```diff
@@ -88,3 +88,204 @@ alt: A screenshot showing how to export a Virtual Machine.
 
 Someone that has access to this file and VirtualBox installed just needs to click `File` in the top left, then `Import` to select that file.
 Once it is imported, they can start the VM as described before, by selecting it from the menu clicking the green start arrow at the top.
+
+(rr-renv-vm-vagrant)=
+## Declarative Virtual Machines with Vagrant
+
+[Vagrant](https://www.vagrantup.com/) is a tool which *"enables users to create and configure lightweight, reproducible, and portable development environments"*.
+In this context, an environment is a virtual machine (its CPUs, RAM, networking and so on) and the machines state (operating system, packages).
+
```

Comment:
```suggestion
Vagrant can set up virtual machines using text scripts, instead of pointing and clicking through a graphical user interface.
This makes it particularly useful for automating the process of setting up virtual machines and making that process reproducible.
```

I _think_ that's what "declarative" means? As a complete newcomer it might help to describe what makes Vagrant different from the VirtualBox that was just described.

Reply:
The way I understand it, declarative is a bit like the opposite of imperative. You specify the state rather than the flow. I could imagine a text format to create a VM that was imperative rather than declarative.

Anyway, I think that extra sentence is good and clarifies how this is different to VirtualBox :+1: